src/Raven.Studio/typescript/viewmodels/database/documents/documents.ts

```diff
@@ -198,9 +202,39 @@ class documents extends viewModelBase {
     }
 
     fetchDocs(skip: number, take: number, previewColumns: string[], fullColumns: string[]): JQueryPromise<pagedResultWithAvailableColumns<any>> {
-        return this.currentCollection().fetchDocuments(skip, take, previewColumns, fullColumns);
+        const continuationTokenToUse = this.getContinuationTokenToUse();
+        
+        return this.currentCollection().fetchDocuments(skip, take, previewColumns, fullColumns, continuationTokenToUse)
+            .done((results => {
+                this.continuationToken = results.continuationToken;
+                // results.continuationToken will be null if there are no more results
+                
+                if (results.continuationToken) {
+                    this.itemsSoFar(this.itemsSoFar() + results.items.length);
+                    
+                    if (this.itemsSoFar() === results.totalResultCount) {
+                        results.totalResultCount = this.itemsSoFar()
+                    } else {
+                        results.totalResultCount = this.itemsSoFar() + 1;
+                    }
+                }
+            }));
     }
 
+    private getContinuationTokenToUse() {
+        const isShardedDatabase = shardedDatabase.isSharded(this.db);
```

Comment:
I wouldn't use that - if server returns continution token then use it. W/o any extra logic.

Reply:
done